### PR TITLE
MISRA-C fixes, switch statement and boolen expression

### PR DESCRIPTION
**Pull request by ceolin**
- Fix switch statement (Rules 16)
- Controlling expressions evaluate Boolean type (Rule 14.4)

### CODE REVIEW

File: boards/x86/galileo/pinmux.c [hunk 1]
```diff
@@ -567,6 +567,9 @@ int _galileo_pinmux_set_pin(struct device *port, u8_t pin, u32_t func)
 		case NONE:
 			/* no need to do anything */
 			break;
+		default:
+			/* shouldn't reach this point */
```
Comment (andrewboie) [inline boards/x86/galileo/pinmux.c]: suggest putting an __ASSERT here

**Review by tyrantlizard77 — APPROVED** [review]

File: boards/x86/galileo/pinmux.c (hunk 1 above)
Comment (ceolin) [reply boards/x86/galileo/pinmux.c]: far enough

**Comment by ceolin** [comment]
#9893 and #9889

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@ceolin please rebase.

File: lib/os/printk.c [hunk 2]
```diff
@@ -122,6 +122,7 @@ void _vprintk(out_func_t out, void *ctx, const char *fmt, va_list ap)
 			case '-':
 				padding = PAD_SPACE_AFTER;
 				goto still_might_format;
+				break;
```
Comment (jhedberg) [inline lib/os/printk.c]: Someone is going to go and remove these statements in the future when enough time has passed that people don't remember why they were added there. Now they look as weird as two consecutive return or break statements. So I'd suggest putting at least a code comment here so people understand that no matter how weird it looks they should not be removed.

Comment (andrewboie) [reply lib/os/printk.c]: I find it baffling that 'goto' isn't an acceptable substitute for 'break' in this context.

Comment (jhedberg) [reply lib/os/printk.c]: Me too. In fact, isn't it likely that compilers will complain about unreachable code in this case?

**Review by nordic-krch — APPROVED** [review]

**Review by jhedberg — DISMISSED** [review]
@ceolin any comment on the break statement issue I raised? At the very least there should be a code comment there to explain it and avoid someone going and removing these in the future.

File: lib/os/printk.c (hunk 2 above)
Comment (ceolin) [reply lib/os/printk.c]: @jhedberg your're right, this happens. The trade-off here is that we'd probably need deviation for dead code for "defensive programming" cases. But I agree with you guys, it is causing more hurt than helping. I'll remove these cases.

**Comment by ceolin** [comment]
@nashif done

**Pull request merged**